### PR TITLE
Restore the `THIRD-PARTY-NOTICES` file

### PR DESCRIPTION
**Pull request by autarch**
Many of the licenses have text saying something to the effect that we must reproduce the license verbatim from the code that the license covers. The SBOM file probably isn't sufficient in those cases.

### CODE REVIEW

File: common.yml [hunk 1]
```diff
@@ -1691,10 +1702,15 @@ tasks:
...
+  - name: check-third-party-notices
+    commands:
+      - func: "fetch source"
+      - func: "check third-party notices"
+
   - name: check-sbom-lite
     commands:
       - func: "fetch source"
-      - func: "check sbom lite"
+      - func: "check third-party notices"
```
Comment (mmcclimon) [inline common.yml:1713]: This change isn't quite right; this should still be "check sbom lite".

File: etc/generate-notices.pl [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+);
+
+print forked_licenses();
+
+for my $entry (sort { $a->[0] cmp $b->[0] or $a->[1] cmp $b->[1] } @license_files) {
+    ( my $package_name = $entry->[0] ) =~ s{vendor/}{};
+    my $file = basename($entry->[1]);
+    my $license_text = do { local ( @ARGV, $/ ) = $entry->[1]; <> };
+    $license_text =~ s/ +$//mg;
+    say "" x 70;
```
Comment (mmcclimon) [inline etc/generate-notices.pl:27]: (I know you didn't write this code, but lol at `say "" x 70`, to be really sure you printed nothing.)

**Review by mmcclimon — APPROVED** [review]
LGTM % the one thing in the evergreen file, but I trust you can do that without needing another review.

(Also, thanks for the throwback to reviewing perl.)

File: common.yml (hunk 1 above)
Comment (autarch) [reply common.yml:1713]: Oops, thanks. I really wish Evergreen config had one _less_ level of structure (the variants -> tasks -> functions -> commands thing is one level too many).

File: etc/generate-notices.pl (hunk 2 above)
Comment (autarch) [reply etc/generate-notices.pl:27]: Heh, I never noticed this.

File: common.yml (hunk 1 above)
Comment (autarch) [reply common.yml]: Fixed.

**Review by ghost — APPROVED** [review]

**Pull request merged**